qat syncbn: add test coverage

vkuzo opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #38820 convbn fusion: add backwards compatibility support
* **#38738 qat syncbn: add test coverage**

Summary:

Adds test coverage for swapping BN -> SyncBN on a fused ConvBN module.

Test Plan:

```
python test/test_quantization.py TestDistributed.test_qat_convbn_fused_syncbn_replacement
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21648320](https://our.internmc.facebook.com/intern/diff/D21648320)

vkuzo committed:
qat syncbn: add test coverage

Summary:

Adds test coverage for swapping BN -> SyncBN on a fused ConvBN module.

Test Plan:

```
python test/test_quantization.py TestDistributed.test_qat_convbn_fused_syncbn_replacement
```

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

raghuramank100 commented on test/quantization/test_workflow_module.py:
```diff
@@ -776,3 +776,37 @@ def test_qat_data_parallel(self):
+            fused_model.qconfig = torch.quantization.get_default_qconfig('fbgemm')
+            torch.quantization.prepare_qat(fused_model, inplace=True)
+            # replace with DDP
+            fused_model = nn.SyncBatchNorm.convert_sync_batchnorm(fused_model)
```
Does the convert logic need to be modified after training ? How about a test to check that convert works with sync bn module?

vkuzo replied:
yep, sounds good.  let's save that for a separate PR?  I don't plan to get to that until we verify SyncBN accuracy with QAT, and it would be great to land this in the meanwhile.

raghuramank100 replied:
Sure, approving it now.

vkuzo committed:
Update on "qat syncbn: add test coverage"


Summary:

Adds test coverage for swapping BN -> SyncBN on a fused ConvBN module.

Test Plan:

```
python test/test_quantization.py TestDistributed.test_qat_convbn_fused_syncbn_replacement
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21648320](https://our.internmc.facebook.com/intern/diff/D21648320)

[ghstack-poisoned]
